plugins/git/git.plugin.zsh

```diff
@@ -1,177 +1,212 @@
 alias gga='git gui citool --amend'
-alias gk='gitk --all --branches'
-
-alias gsts='git stash show --text'
-alias gsta='git stash'
-alias gstp='git stash pop'
-alias gstd='git stash drop'
-
-# Will cd into the top of the current repository
-# or submodule.
-alias grt='cd $(git rev-parse --show-toplevel || echo ".")'
+ggf() {
+[[ "$#" != 1 ]] && b="$(current_branch)"
+git push --force origin "${b:=$1}"
+}
+compdef _git ggf=git-checkout
+ggl() {
+[[ "$#" != 1 ]] && b="$(current_branch)"
+git pull origin "${b:=$1}"
+}
+compdef _git ggl=git-checkout
+ggp() {
+[[ "$#" != 1 ]] && b="$(current_branch)"
+git push origin "${b:=$1}"
+}
+compdef _git ggp=git-checkout
+ggpnp() {
+ggl "$1" && ggp "$1"
+}
+compdef _git ggpnp=git-checkout
+alias ggsup='git branch --set-upstream-to=origin/$(current_branch)'
+ggu() {
+[[ "$#" != 1 ]] && b="$(current_branch)"
+git pull --rebase origin "${b:=$1}"
+}
+compdef _git ggu=git-checkout
 
-# Git and svn mix
+alias gignore='git update-index --assume-unchanged'
+alias gignored='git ls-files -v | grep "^[[:lower:]]"'
 alias git-svn-dcommit-push='git svn dcommit && git push github master:svntrunk'
 compdef git-svn-dcommit-push=git
 
-alias gsr='git svn rebase'
-alias gsd='git svn dcommit'
-#
-# Will return the current branch name
-# Usage example: git pull origin $(current_branch)
-#
-function current_branch() {
-  ref=$(git symbolic-ref HEAD 2> /dev/null) || \
-  ref=$(git rev-parse --short HEAD 2> /dev/null) || return
-  echo ${ref#refs/heads/}
-}
-
-function current_repository() {
-  ref=$(git symbolic-ref HEAD 2> /dev/null) || \
-  ref=$(git rev-parse --short HEAD 2> /dev/null) || return
-  echo $(git remote -v | cut -d':' -f 2)
-}
+alias gk='\gitk --all --branches'
+compdef _git gk='gitk'
+alias gke='\gitk --all $(git log -g --pretty=format:%h)'
+compdef _git gke='gitk'
 
-# these aliases take advantage of the previous function
-alias ggpull='git pull origin $(current_branch)'
-compdef ggpull=git
-alias ggpur='git pull --rebase origin $(current_branch)'
-compdef ggpur=git
-alias ggpush='git push origin $(current_branch)'
-compdef ggpush=git
-alias ggpnp='git pull origin $(current_branch) && git push origin $(current_branch)'
-compdef ggpnp=git
-
-# Pretty log messages
-function _git_log_prettily(){
-  if ! [ -z $1 ]; then
-    git log --pretty=$1
-  fi
-}
+alias gl='git pull'
+alias glg='git log --stat --color'
+alias glgp='git log --stat --color -p'
+alias glgg='git log --graph --color'
+alias glgga='git log --graph --decorate --all'
+alias glgm='git log --graph --max-count=10'
+alias glo='git log --oneline --decorate --color'
+alias glol="git log --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%cr) %C(bold blue)<%an>%Creset' --abbrev-commit"
+alias glola="git log --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%cr) %C(bold blue)<%an>%Creset' --abbrev-commit --all"
+alias glog='git log --oneline --decorate --color --graph'
+alias glone='git clone'
```

Comment:
For `git clone` `gcl` seems to be better.

I currently use this function for it:

    gcl () {
      git clone --recursive $@ || return
      local last=$@[$#]
      if [[ -d $last ]]
      then
        cd $last
      else
        local dir=${${last:t}%.git}
        if [[ -d $dir ]]
        then
          cd $dir
        fi
      fi
    }

When using just an alias, it should have `--recursive`, too.